src/dev/flang/util/List.java

```diff
@@ -277,6 +277,39 @@ public String toString(String pre, String sep, String term, Function<T, String>
   }
 
 
+  /**
+   * add adds given element to the list
+   *
+   * @param e element to add
+   */
+  @Override
+  public boolean add(T e)
+  {
+    if (ANY.PRECONDITIONS) ANY.require
+      (true || !isFrozen() // NYI: disabled since tests/reg_issue1943_type_parameter_as_outer_type crashes if enabled, need to check
+       );
+
+    return super.add(e);
+  }
+
+
+  /**
+   * add adds given element to the list after possibly cloning the list in case
+   * it was frozen.
+   *
+   * @param e element to add
+   *
+   * @return this if !isFrozen(), a clone of this otherwise, in any case with e
+   * added.
+   */
+  public List<T> addAfterUnfreeze(T e)
```

Comment:
I stumbled here while trying to understand what the purpose of this is. I think this needs an explanation why this is done like this.